src/eos-crash-metrics.c

```diff
@@ -137,6 +145,111 @@ normalize_path (char *path)
     path[i] = path[i] == '!' ? '/' : path[i];
 }
 
+static GPtrArray *
+get_installed_ref_for_executable (FlatpakInstallation *installation,
+                                  const char *executable_name,
+                                  GError **error)
+{
+  g_autoptr(GPtrArray) xrefs = NULL;
+  GPtrArray *applications = NULL;
+  FlatpakInstalledRef *xref = NULL;
+  guint i;
+
+  /* get apps and runtimes */
+  xrefs = flatpak_installation_list_installed_refs (installation,
+                                                    NULL, error);
+  if (xrefs == NULL)
+    return NULL;
+
+  applications = g_ptr_array_new ();
+
+  for (i = 0; i < xrefs->len; i++)
+    {
+      g_autofree gchar *executable_path = NULL;
+      FlatpakInstalledRef *xref = g_ptr_array_index (xrefs, i);
+
+      /* only apps */
+      if (flatpak_ref_get_kind (FLATPAK_REF (xref)) != FLATPAK_REF_KIND_APP)
+        continue;
+
+      executable_path = g_build_filename (flatpak_installed_ref_get_deploy_dir (FLATPAK_INSTALLED_REF (xref)),
+                                          "files",
+                                          "bin",
+                                          executable_name,
+                                          NULL);
+      /* found a Flatpak with the same application name */
+      if (g_file_test (executable_path, G_FILE_TEST_EXISTS))
+        g_ptr_array_add (applications, g_object_ref (xref));
```

Comment:
I think it would be acceptable to return directly the first ref that matches from this function.